mitodl/open-discussions
# Added xpro integration for catalog

#### Pre-Flight checklist


#### What are the relevant tickets?

Fixes #2182 

#### What's this PR do?

Adds ETL code to pull in xPro program/course catalog, but this involved a bevy of other changes to support this:

- Added loading functions for course runs, prices, topics, and instructors
- Added called to index programs and courses
- Additional fields added to `Program`
- Added `prices` relation to `Program`
- Updated indexing code to *not* index programs, courses, course runs with `published=False`
- Removed the hardcoded ES filter to not return `published=False` results since they are no longer indexed at all
- Updated indexing code to provide an upsert operation for programs and courses (this could be ported to other types, but kept it to scoped to those necessary for this issue)
- Adjustments to some of our catalog factories to support some use cases

#### How should this be manually tested?

- In xPro, checkout the branch `nl/progams_public_api`
- In xPro, ensure you have a mix of programs and courses with and without prices
- Run the `course_catalog.tasks.get_xpro_data` task and verify it runs and the data looks reasonable. Particularly, programs and courses without prices should not appear.
